tests/CMakeLists.txt

```diff
@@ -1,12 +1,19 @@
 if(UNIX)
-    catkin_add_gtest(${PROJECT_NAME}-test unix_serial_tests.cc)
-    target_link_libraries(${PROJECT_NAME}-test ${PROJECT_NAME})
+    find_package(GTest REQUIRED)
+    include_directories(${GTEST_INCLUDE_DIRS})
```

Comment:
https://cmake.org/cmake/help/latest/module/FindGTest.html

Please depend on GTest via its imported targets rather than the old school variable-based approach. The target you'll want to use here is `GTest::GTest`. That also lets you remove the use of `${GTEST_LIBRARIES}`.

Reply:
One advantage to doing what Chis is recommending here is that because CMake is stringly-typed like most shells if the variable `GTEST_LIBRARIES` doesn't exist because the find_package failed, it will equal an empty string. This means you'll get a much more confusing error than if you use the target-based approach. @ChrisThrasher is this a correct way to understand why someone should bias towards the target based approach?

Reply:
That is one amongst various good reason to prefer `IMPORTED` targets over variable expansion.

Reply:
:+1: made this change https://github.com/wjwwood/serial/pull/285/commits/f60d49d91b0b58a2427a082e3e9ef08db8d8398d